modules/mono/glue/GodotSharp/GodotSharp/Core/IO/File.cs

```diff
@@ -0,0 +1,571 @@
+
+            return new FileStream(file, access);
+        }
+
+        public static FileStream OpenEncryptedWithPass(string path, FileAccess access, string pass)
+        {
+            ThrowIfParamPathIsNullOrEmpty(path, nameof(path));
+
+            if (pass == null)
+                throw new ArgumentNullException(nameof(pass), "Password cannot be null.");
+            if (pass.Length == 0)
+                throw new ArgumentException("Password cannot be empty.", nameof(pass));
+
+            var file = new File();
+
+            file.OpenEncryptedWithPass(path, FileAccessToModeFlags(access), pass)
+                .DisposeAndThrowOnError(file);
+
+            return new FileStream(file, access);
+        }
+
+        public static DateTime GetCreationTime(string path)
+        {
+            using var file = new File();
+            var dateTime = DateTimeOffset.FromUnixTimeSeconds((long)file.GetModifiedTime(path));
+            return dateTime.LocalDateTime;
+        }
+
+        public static DateTime GetCreationTimeUtc(string path)
+        {
+            using var file = new File();
+            var dateTime = DateTimeOffset.FromUnixTimeSeconds((long)file.GetModifiedTime(path));
+            return dateTime.UtcDateTime;
+        }
+
+        public static FileStream OpenRead(string path) => Open(path, FileAccess.Read);
+
+        public static FileStream OpenWrite(string path) => Open(path, FileAccess.Write);
+
+        public static string ReadAllText(string path) => ReadAllText(path, Encoding.UTF8);
+
+        public static string ReadAllText(string path, Encoding encoding)
+        {
+            ThrowIfParamPathIsNullOrEmpty(path, nameof(path));
+
+            if (encoding == null)
+                throw new ArgumentNullException(nameof(encoding));
+
+            using var stream = new StreamReader(Open(path, FileAccess.Read), encoding, detectEncodingFromByteOrderMarks: true);
+            return stream.ReadToEnd();
+        }
+
+        public static void WriteAllText(string path, string contents)
+        {
+            ThrowIfParamPathIsNullOrEmpty(path, nameof(path));
+
+            using var stream = new StreamWriter(Open(path, FileAccess.Write));
+            stream.Write(contents);
+        }
+
+        public static void WriteAllText(string path, string contents, Encoding encoding)
+        {
+            ThrowIfParamPathIsNullOrEmpty(path, nameof(path));
+
+            if (encoding == null)
+                throw new ArgumentNullException(nameof(encoding));
+
+            using var stream = new StreamWriter(Open(path, FileAccess.Write), encoding);
+            stream.Write(contents);
+        }
+
+        public static byte[] ReadAllBytes(string path)
+        {
+            using var stream = Open(path, FileAccess.Read);
+
+            long length = stream.Length;
+
+            if (length > int.MaxValue)
+                throw new IOException("The file is too long. Only files less than 2 GBs in size are supported.");
```

Comment:
What is preventing us from supporting 64-bit file access? I think we should. IIRC Godot will support this in core in 4.0.

Reply:
Array max length is `int.MaxValue` (~2 GB). The purpose of this method is to mimic `System.IO.File.ReadAllBytes` which returns a `byte[]`, so we're limited to that.
It's not a good idea to read that much at once any way. Better read in chunks, even for much smaller files.